docs/express-vs-core.qmd

```diff
@@ -0,0 +1,262 @@
+- Implicit vs. explicit placement of outputs
+- Different syntax for UI containers
+
+Let's examine each of these in more detail.
+
+### Import statements
+
+A Shiny Core app file usually contains an import statement like:
+
+```python
+from shiny import App, reactive, render, ui
+```
+
+In Shiny Express, you'll instead see:
+
+```python
+from shiny import reactive
+from shiny.express import input, render, ui
+```
+
+Note that both import `ui` and `render`, but from different places.
+While Express's `shiny.express.ui` has almost all of the same UI functions as Core's `shiny.ui`, their function signatures often differ slightly, to reflect Express's different usage patterns.
+And the `render` functions---well actually, they are identical right now, but we're planning to add some Express-specific features to the `shiny.express.render` versions in the near future.
+
+Meanwhile, the `reactive` module is unchanged between Core and Express, as the two modes share the same reactive foundations.
+
+Finally, notice that Express also imports `input` from `shiny.express`.
+This isn't needed in Core, because the `input` object is passed into the server function as an argument.
+Since Express has no server function, we made it an attribute of `shiny.express`.
+
+### Organization of UI and server code
+
+**Every Shiny Core app file has a UI section, and a server section;** the two are separate and distinct.
+
+The UI section consists of a single (potentially very long and deeply nested) expression, stored as a variable named `app_ui` by convention.
+The object this produces is actually straight up HTML, which is sent to the browser when it first loads the app.
+
+The server section is a function, named `server` by convention, that always takes the arguments `input`, `output`, and `session`.
+This function contains render functions and reactive functions, which are used to update the UI in response to user input.
+
+You can think of the UI section as a template, with `ui.output_xx()` calls as placeholders for outputs, and the server section as the instructions for how to populate those outputs.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+# Core
+from shiny import ui, render, reactive, App
+from datetime import datetime
+
+app_ui = ui.page_fixed(
+    ui.h1("Title"),
+    ui.output_code("greeting"),
+)
+
+def server(input, output, session):
+    @reactive.Calc
+    def time():
+        reactive.invalidate_later(1)
+        return datetime.now()
+
+    @render.code
+    def greeting():
+        return f"Hello, world!\nIt's currently {time()}."
+
+app = App(app_ui, server)
+```
+
+**In Shiny Express, there isn't this hard distinction** between UI and server.
+
+Instead, everything coexists as top-level code in the app file: input components, layout directives, outputs (including their rendering logic), and reactive functions.
+
+Also, you aren't forced to combine your UI into a single complex object.
```

Comment:
I would cut this, as part of our pitch for Core is that the UI object is simpler to read and maintain, aslo we say you can combine variables into a UI object. IMO it's enough just say that everything coexists.

Reply:
It is simpler to read and maintain, and harder to write. Putting it in a single object is a downside--I felt it viscerally yesterday when porting the dashboard app.